packages/features/handleMarkNoShow.ts

```diff
@@ -0,0 +1,126 @@
+  if (failed.length < 1) return;
+  const failedMessage = failed.map((r) => r.reason);
+  console.error("Failed to update no-show status", failedMessage.join(","));
+};
+
+const handleMarkNoShow = async ({ bookingUid, attendees }: TNoShowInputSchema) => {
+  try {
+    const attendeeEmails = attendees?.map((attendee) => attendee.email) || [];
+    if (attendees && attendeeEmails.length > 0) {
+      const allAttendees = await prisma.attendee.findMany({
+        where: {
+          AND: [
+            {
+              booking: {
+                uid: bookingUid,
+              },
+              email: {
+                in: attendeeEmails,
+              },
+            },
+          ],
+        },
+        select: {
+          id: true,
+          email: true,
+        },
+      });
+
+      const allAttendeesMap = allAttendees.reduce((acc, attendee) => {
+        acc[attendee.email] = attendee;
+        return acc;
+      }, {} as Record<string, { id: number; email: string }>);
+
+      const updatePromises = attendees.map((attendee) => {
+        const attendeeToUpdate = allAttendeesMap[attendee.email];
+        if (!attendeeToUpdate) return;
+        return prisma.attendee.update({
+          where: { id: attendeeToUpdate.id },
+          data: { noShow: attendee.noShow },
+        });
+      });
+
+      const results = await Promise.allSettled(updatePromises);
+      logFailedResults(results);
+
+      const _attendees = results
+        .filter((x) => x.status === "fulfilled")
+        .map((x) => (x as PromiseFulfilledResult<{ noShow: boolean; email: string }>).value)
+        .map((x) => ({ email: x.email, noShow: x.noShow }));
+
+      const payload = await getResultPayload(_attendees);
+      const booking = await prisma.booking.findUnique({
+        where: { uid: bookingUid },
+        select: {
+          id: true,
+          eventType: {
+            select: {
+              id: true,
+              teamId: true,
+              userId: true,
+            },
+          },
+        },
+      });
+      const orgId = await getOrgIdFromMemberOrTeamId({
+        memberId: booking?.eventType?.userId,
+        teamId: booking?.eventType?.teamId,
+      });
+      const webhooks = await new WebhookService({
+        teamId: booking?.eventType?.teamId,
+        userId: booking?.eventType?.userId,
+        eventTypeId: booking?.eventType?.id,
+        orgId,
+        triggerEvent: WebhookTriggerEvents.BOOKING_NO_SHOW_UPDATED,
+      });
+
+      const t = await getTranslation("en", "common");
+      await webhooks.sendPayload({
+        ...payload,
+        /** We send webhook message pre-translated, on client we already handle this */
+        // @ts-expect-error payload is too booking specific, we need to refactor this
+        message: t(payload.message, { x: payload.attendees[0]?.email || "User" }),
+        bookingUid,
+        bookingId: booking?.id,
+      });
+      return payload;
```

Comment:
We are updating either attendees `noShow` or `hostNoShow` in an API call. Through frontend I think it was perfectly fine as user takes action one by one but through API, I believe developers would want it to be done in same call soon.